### PR TITLE
[librdkafka] Upgrade up to 2.1.1

### PR DESCRIPTION
**Pull request by alexander-smyslov**
- [X] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md)
- [X] SHA512s are updated for each updated download
- [ ] ~~The "supports" clause reflects platforms that may be fixed by this new version~~
- [ ] ~~Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.~~
- [X] Any patches that are no longer applied are deleted from the port's directory.
- [X] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [X] Only one version is added to each modified port's versions file.

### CODE REVIEW

**Comment by dg0yt** [comment]
Does this update install valid pkgconfig files?

I recently noticed (while reviewing cppkafka PR):
> Both Libs.private and Requires.private are exclusively for static linkage, for additonal link libraries.
> IIUC rdkafka got it wrong in two ways: By installing different names for different linkage (inconvenient for downstream usage, as in this port), and by publishing additional link libraries only for dynamic linkage (when this information is already carried by DLLs/shared objects).

File: ports/librdkafka/portfile.cmake [hunk 1]
```diff
@@ -1,12 +1,11 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO edenhill/librdkafka
-    REF v2.1.0
-    SHA512 eeefa8833e7c178015ff8d0914d927d1bf4e09a75c5f227ba75e527f9bfc46407798d69642b5db3edafb93fcdbe56e8b506df826d01c7793af0f11e96c38caed 
+    REF v2.1.1
```
Comment (jimwang118) [inline ports/librdkafka/portfile.cmake:4]:
```suggestion
    REF "v${VERSION}"

```

**Review by jimwang118 — CHANGES_REQUESTED** [review]

**Comment by jimwang118** [comment]
Note: I will be converting your PR to draft status. The above suggested changes are only recommendations. If you are willing to adopt them, please click "ready for review" after making the modifications. If you do not wish to make any changes, please click "ready for review" directly. That way, I can be aware that you've responded since you can't modify the tags.

**Review by jimwang118 — DISMISSED** [review]

File: ports/librdkafka/portfile.cmake [hunk 2]
```diff
@@ -1,12 +1,12 @@
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO edenhill/librdkafka
-    REF v2.1.0
-    SHA512 eeefa8833e7c178015ff8d0914d927d1bf4e09a75c5f227ba75e527f9bfc46407798d69642b5db3edafb93fcdbe56e8b506df826d01c7793af0f11e96c38caed 
+    REF "v${VERSION}"
+
```
Comment (jimwang118) [inline ports/librdkafka/portfile.cmake:5]: There is a blank line here, it is recommended to delete it.

Comment (alexander-smyslov) [reply ports/librdkafka/portfile.cmake]: done

**Comment by jimwang118** [comment]
The usage and feature has been tested successfully locally.
```
find_package(RdKafka CONFIG REQUIRED)
target_link_libraries(main PRIVATE RdKafka::rdkafka RdKafka::rdkafka++)
```

**Pull request merged**